config/initializers/feature_flags.rb

```diff
@@ -0,0 +1,11 @@
+FEATURES = {
+  basic_auth: {
+    enabled:  Rails.env.production? || (!ENV['BASIC_AUTH_FORCE'].blank? && ENV['BASIC_AUTH_FORCE'] != 'false'),
+    username: ENV['BASIC_AUTH_USERNAME'],
```

Comment:
There's a bit of mixing up going on in this file. It's dealing with feature flags but it's also configuring the basic auth credentials. Could they be split?

Reply:
So, the main reason I decided to add this is now instead of using env vars directly is for testing purposes. With direct env vars access, I'd have to stub ENV and its parts to check if basic auth is required in the right places, which feels messy. This way, there's one place where feature toggling and the relevant config lives. Could renaming this to features.rb make things better?

Reply:
This becomes basic_auth.rb to make it more specific. It may morph into feature flags in the future.